agents/web-api/src/main/java/edu/jhuapl/dorset/agents/StockAgent.java

```diff
@@ -0,0 +1,228 @@
+        this.client = client;
+        this.apiKey = apiKey;
+
+        ICsvBeanReader csvBeanReader = null;
+
+        InputStream nasdaqCompanies = StockAgent.class.getClassLoader().getResourceAsStream("stockagent/NASDAQ_Companies.csv");
+
+        this.stockSymbolMap = new HashMap<String, String>();
+        try {
+            csvBeanReader = new CsvBeanReader(new BufferedReader( new InputStreamReader(nasdaqCompanies)),
+                    CsvPreference.STANDARD_PREFERENCE);
+            final String[] header = csvBeanReader.getHeader(true);
+            CompanyInfo companyInfo;
+            while ((companyInfo = csvBeanReader.read(CompanyInfo.class, header, processors)) != null) {
+                this.stockSymbolMap.put(companyInfo.getName(), companyInfo.getSymbol());
+            }
+
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        } finally {
+            if (csvBeanReader != null) {
+                try {
+                    csvBeanReader.close();
+                } catch (IOException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+
+                }
+            }
+        }
+
+        InputStream nyseCompanies = StockAgent.class.getClassLoader().getResourceAsStream("stockagent/NYSE_Companies.csv");
+
+        try {
+            csvBeanReader = new CsvBeanReader(new BufferedReader( new InputStreamReader(nyseCompanies)),
+                    CsvPreference.STANDARD_PREFERENCE);
+            final String[] header = csvBeanReader.getHeader(true);
+            CompanyInfo companyInfo;
+            while ((companyInfo = csvBeanReader.read(CompanyInfo.class, header, processors)) != null) {
+
+                if (!this.stockSymbolMap.containsKey("apple")) {
+                    this.stockSymbolMap.put(companyInfo.getName(), companyInfo.getSymbol());
+                }
+            }
+
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        } finally {
+            if (csvBeanReader != null) {
+                try {
+                    csvBeanReader.close();
+                } catch (IOException e) {
+                    // TODO Auto-generated catch block
+                    e.printStackTrace();
+                }
+            }
+        }
+
+    }
+
+    @Override
+    public AgentResponse process(AgentRequest request) {
+
+        String requestCompanyName = request.getText().toLowerCase().replace("stocks ", "").replace("stock ", "");
+        String[] stockCompanyInfo = findStockSymbol(requestCompanyName);
+        String keywordCompanyName = stockCompanyInfo[0];
+        String keywordCompanySymbol = stockCompanyInfo[1];
+        String json = null;
+
+        if (keywordCompanySymbol == null) {
+            return new AgentResponse("I am sorry, I don't understand which company you are asking about.");
```

Comment:
Added a response code for cases where the agent did not know the answer because it could not identify a company name (ResponseStatus.Code.AGENT_DID_NOT_KNOW_RESPONSE)